### ISSUE TITLE
confused about rustc E0223, ambiguous associated type

### ISSUE THREAD

**Issue by youweol-master** [rust-lang/rust#106029]
here is a small snippet
```rust 
pub trait A {
   type AItem: Debug + Default;
   fn g() {}
}

pub trait B {
   type BItem: A;

   fn f(x: Self::BItem::AItem) {       // this line raise E0223
       println!("{}", x);
   }

   fn gg() {
       let x: Self::BItem::AItem = Default::default();    // raise E0223 too
       Self::BItem::g();
   }
}
```

what I'm confused is why  inside B's gg()
```
Self::BItem::g();
```
did't raise E0223.

According to the rustc error code document, code like
```
Self::BItem::AItem
```
need to be fully-qualified syntax, because Self::BItem is just a placeholder.
But why
```
Self::BItem::g();
```
can success compile, which Self::BItem have same situation too.

writting `<<Self as B>::BItem as A>::AItem` is too much  verbose than Self::BItem::AItem.

Inside trail B ,the rustc is smart enough to recognized BItem is a type of A because of
```
   type BItem: A;
```
so `Self::BItem::g()`  is equal to `<<Self as B>::BItem as A>::g()`.
But why did it blind to see that 
```
Self::BItem::AItem
```
the Self::BItem must be a type of A.
